### ISSUE TITLE
Numeric input fields show standard OSK on (initial) focus

### ISSUE THREAD

**Issue by tarek-y-ismail** [canonical/mir#3696]
Related? #3580

When focusing on numeric input fields, `TextInputV?::set_content_type` (checked V1 and V3) successfully sets the hint and purpose, yet the OSK that pops up is  the standard character/numeric OSK.  This occurs with both the [iot example](https://github.com/canonical/iot-example-graphical-snap) (Electron, TextInputV1), and [WPE webkit](https://snapcraft.io/wpe-webkit-mir-kiosk) (TextInputV3). This also occurs with the [w3 schools HTML example](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_number), and [Inductive Automation's Ignition demo](https://demo.inductiveautomation.com/data/perspective/client/OnlineDemo/feature/reporting/examples)

Reproduction steps:
1. Get the iot-example or WPE webkit, and ubuntu-frame-osk
2. Launch one of the URLs mentioned above
  - To change the URL for the iot example: Open `snap/snapcraft.yaml`, replace the url inside `loadURL`
  - To change the URL for WPE webket: `sudo snap set wpe-webkit-mir-kiosk url=<url>`
3. Launch miral-app, the iot example, and ubuntu-frame-osk:
  - `WAYLAND_DISPLAY=wayland-99 ./path/to/miral-app --add-wayland-extensions=all`
  - `WAYLAND_DISPLAY=... iot-graphical-example`
  - `WAYLAND_DISPLAY=... ubuntu-frame-osk`
4. Click on the number field

Expected: A numerical keyboard pops up
Reality: The standard keyboard pops up

**Comment by Saviq** [comment]
Is the content type forwarded to the OSK on the input method protocol?

We really need to get maliit working so we can have a comparison…

**Comment by tarek-y-ismail** [comment]
> Is the content type forwarded to the OSK on the input method protocol?

Yep. I set a breakpoint in `TextInputV1::set_content_type` and it sure is set, but I didn't follow it around.

> We really need to get maliit working so we can have a comparison…

I think I got that working yesterday under GNOME. It just launches now

**Comment by Saviq** [comment]
> Yep. I set a breakpoint in TextInputV1::set_content_type and it sure is set, but I didn't follow it around.

That's the client side, it's the input-method-unstable-v2 protocol that Mir then communicates with the OSK software - and forwards client requests to. If the content type hint gets lost in Mir, we've to fix that. But if it correctly gets forwarded to the OSK, the issue is there.

As we know, it does at least work _sometimes_ with the numerical input fields, so there is a question about where does it get lost when it doesn't work.

**Comment by tarek-y-ismail** [comment]
Solved by #3681
